plux/build/setuptools.py

```diff
@@ -107,6 +109,46 @@ def _ensure_dist_info(self):
 patch_editable_wheel_command()
 
 
+def patch_egg_info_command():
+    """
+    This patch fixes the build process when pip builds a wheel from a source distribution. A source distribution built
+    with plux will already contain an `.egg_info` directory, however during the build with pip, a new .egg-info
+    directory is created from scratch from the python distribution configuration files (like setup.cfg or
+    pyproject.toml). This is a problem, as building a wheel involves creating a .dist-info dir, which is populated from
+    this new .egg-info directory. The .egg-info shipped with the source distribution is completely ignored during this
+    process, including the `plux.json`, leading to a wheel that does not correctly contain the `entry_points.txt`.
+
+    This patch hooks into this procedure, and makes sure when this new .egg-info directory is created, we first locate
+    the one shipped with the source distribution, locate the `plux.json` file, and copy it into the new build context.
+    """
+    _run_orig = egg_info.run
+
+    def _run(self):
+        LOG.debug("Running egg_info command patch from plux")
+        # the working directory may be something like `/tmp/pip-req-build-bwekzpi_` where the source distribution
+        # was copied into. this happens when you `pip install <dist>` where <dist> is
+        # some source distribution.
+        should_read, meta_dir = _should_read_existing_egg_info()
+
+        # if the .egg_info from the source distribution contains the plux file, we prepare the new .egg-info by copying
+        # it there. everything else should be handled implicitly by the command
+        if should_read:
+            LOG.debug("Locating plux.json from local build context %s", meta_dir)
+            plux_json = os.path.join(meta_dir, "plux.json")
+            if os.path.exists(plux_json):
+                self.mkpath(self.egg_info)  # this is what egg_info.run()
```

Comment:
nit: It seems like this comment didn't get to the end: "# this is what egg_info.run()"
I think you are referring to this line in setuptools, right?
https://github.com/pypa/setuptools/blob/e5f16a2a990ff18cdef27a22b742f97444867186/setuptools/command/egg_info.py#L297